File: Course-3/Mobile-dev/Task_2/Base/app/src/main/java/org/hse/android/MainActivity.java
package org.hse.android;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Objects.requireNonNull(getSupportActionBar()).hide();

        View button_student = findViewById(R.id.button_student);
        View button_teacher = findViewById(R.id.button_teacher);
        View button_settings = findViewById(R.id.button_settings);

        button_student.setOnClickListener(v -> {
            //Intent intent = new Intent(MainActivity.this, StudentActivity.class);
            //startActivity(intent);
            Student_pressed();
        });
        button_teacher.setOnClickListener(v -> {
            //Intent intent = new Intent(MainActivity.this, TeacherActivity.class);
            //startActivity(intent);
            Teacher_pressed();
        });
        button_settings.setOnClickListener(v -> {
            //Intent intent = new Intent(MainActivity.this, StudentActivity.class);
            //startActivity(intent);
            Settings_pressed();
        });
    }

    private void Student_pressed(){
        Intent intent = new Intent(MainActivity.this, StudentActivity.class);
        startActivity(intent);
    }

    private void Teacher_pressed(){
        Intent intent = new Intent(MainActivity.this, TeacherActivity.class);
        startActivity(intent);
    }

    private void Settings_pressed(){
        Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
        startActivity(intent);
    }
}

File: Course-3/Mobile-dev/Stepik/6_activeui/6-1/6-1-12/app/src/main/java/com/example/company/myapplication/MainActivity.kt
package com.example.company.myapplication

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener{
            val myIntent = Intent(this, SecondActivity::class.java)
            myIntent.putExtra("info", editText.text.toString())
            startActivity(myIntent)
        }
        editText.setText("")
        textView.text = intent.getStringExtra("info")
    }

    override fun onResume() {
        super.onResume()
        editText.setText("")
        textView.text = intent.getStringExtra("info")
    }
}